src/Sulu/Bundle/AdminBundle/Resources/js/components/Radio/Radio.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import * as React from 'react';
+import radioStyles from './radio.scss';
+
+type Props = {
+    checked: boolean,
+    value: string,
+    onChange?: (value: string) => void,
+};
+
+export default class Radio extends React.PureComponent<Props> {
+    static defaultProps = {
+        value: 'on',
+        checked: false,
+    };
+
+    input: React.ElementRef<'input'>;
+
+    handleChange = () => {
+        if (this.props.onChange) {
+            this.props.onChange(this.props.value);
+        }
+    };
+
+    handleButtonClick = () => this.input.click();
+    handleInputClick = (event: Event) => event.stopPropagation();
+    setInput = (input: React.ElementRef<'input'>) => this.input = input;
+
+    render() {
+        return (
+            <span onClick={this.handleButtonClick} className={radioStyles.radio}>
+                <input
+                    ref={this.setInput}
+                    type="radio"
+                    checked={this.props.checked}
+                    onClick={this.handleInputClick}
+                    onChange={this.handleChange} />
+                <span />
+            </span>
```

Comment:
And do we need the spans at all? We are just returning a single element here. Or are you using the outer div for styling and you are hiding the radio itself?